### PR TITLE
feat(TTD): Filter tasks based on due dates

### PR DESCRIPTION
**Pull request by mohammed-sharraf-sazim**
Description:
- Added the filtering option where users can filter based on due dates (Due Today, Upcoming, Overdue).

Associated Ticket Link(s):
- https://trello.com/c/fMZPIQtz/11-filter-tasks-due-dates

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -11,6 +11,7 @@ function App() {
   const [tasks, setTasks] = useState<Tasks[]>([]);
   const [filter, setFilter] = useState<string>('all');
   const [priorityFilter , setPriorityFilter] = useState<string>('all');
+  const [dateFilter, setDateFilter] = useState<string>('all');
```
Comment (nur-zaman-sazim) [inline src/App.tsx:14]: use Enum for date filters

File: src/App.tsx [hunk 2]
```diff
@@ -35,19 +36,34 @@ function App() {
...
+      return task.priority === priorityFilter && dateCondition;
     } else if (filter === 'completed' && priorityFilter !== 'all') {
-      return task.isCompleted && task.priority === priorityFilter as Priority;
+      return task.isCompleted && task.priority === priorityFilter  && dateCondition;
     } else if (filter === 'active' && priorityFilter !== 'all') {
-      return !task.isCompleted && task.priority === priorityFilter as Priority;
+      return !task.isCompleted && task.priority === priorityFilter && dateCondition;
     }
     return false;
   });
```
Comment (nur-zaman-sazim) [inline src/App.tsx:69]: Better to move the logics in a separate file

**Review by nur-zaman-sazim — CHANGES_REQUESTED** [review]

**Review by mMohsinSazim — CHANGES_REQUESTED** [review]
Why The comments are not addressed yet??

File: src/App.tsx (hunk 1 above)
Comment (mohammed-sharraf-sazim) [reply src/App.tsx:14]: If I use enum I have to make some major changes.

File: src/App.tsx (hunk 2 above)
Comment (mohammed-sharraf-sazim) [reply src/App.tsx:69]: This changed require some major refactoring

Comment (mohammed-sharraf-sazim) [reply src/App.tsx]: Created a separate file for the logic

**Review by nur-zaman-sazim — APPROVED** [review]

**Pull request merged**